src/algorithms/GenericFunctions.jl

```diff
@@ -4,6 +4,152 @@
 #
 ###############################################################################
 
+function ^(a::T, n::Integer) where T <: RingElem
+   n < 0 && throw(DomainError(n, "exponent must be >= 0"))
```

Comment:
Maybe try `inv`? We are a bit inconsistent here. `fmpz(1)^-1` works, but `one(QQ["x"][1])` fails.

Reply:
One of the problems for ^ and powmod is that the unary - doesn't work on `typemin(Int)`. and then `unsigned(Int)` works but `unsigned(BigInt)` doesn't. :(